src/NUnitFramework/framework/Internal/HookExtensions/HookExtension.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Linq;
+
+namespace NUnit.Framework.Internal.HookExtensions
+{
+    /// <summary>
+    /// Hook Extension interface to run custom code synchronously before or after any test activity.
```

Comment:
The class is not an interface, at least not in the C# sense.  Having looked at it, I am not sure what it really is.  

The TestHook class holds the eventhandlers.  
The HookExtension holds two TestHooks, Before and After, for wrapping the test to be run (or fixture or whatever I assume).   I can't see that this can be called an interface.